### PR TITLE
Use Pexpect on maverick to auto enter pass

### LINKED ISSUE #3: Use Pexcept as a workaround for entering SSH password everytime
[`wip/pexpect`](https://github.com/sibanez12/mpi_bcp_qos_sim/tree/wip/pexpect) seems to be working, with the exception the there is some funny issues with printing to stdout from the spawned threads. We suspect it's something to do with sharing stdout, or something to do with the buffers in the [Pexcept.spawn](http://pexpect.readthedocs.io/en/stable/api/pexpect.html#spawn-class) object. For example, changing the `maxread` parameter had some effect, but didn't solve our problem.

Ultimately, it seems clear that the client and server code that we expect to be called is indeed called. We just don't have nice ways to print that at the moment.

### PR DESCRIPTION
**Pull request by lukehsiao**
Closes #3. Just fill in the `KEY` variable in `run_sim.py`. But, **never** commit your plaintext pass!

Just opening this PR for now. We can vet it more fully before merging, or update this to use another file rather than a variable, but it seems to work so far.

### CODE REVIEW

File: run_sim.py [hunk 1]
```diff
@@ -6,27 +7,37 @@
 from time import sleep
 from StatsParser import StatsParser
 from StatsAnalysis import *
+import pexpect
 
 SIM_RUN_TIME = 30
 OUTPUT_DIR = "./out/"
 BUILD = "./Debug"
 SIM_LOGGING_TIME = 5
+KEY = "fill_in_pass"
```
Comment (lukehsiao) [inline run_sim.py]: Could move this out to a separate file, and just read from that file. That file could then be left out of the repo, and we wouldn't have to worry about someone accidentally committing sensitive info.

File: run_sim.py [hunk 2]
```diff
@@ -6,27 +7,37 @@
...
+        print "[info] sending " + str(signal.SIGUSR1) + " to pid: " + str(process.pid)
+        process.kill(signal.SIGUSR1)
     sleep(SIM_LOGGING_TIME) # wait for the sim to finish up and write results
     print "killing process"
-    process.terminate()
-
-    # process.wait()
+    process.close()
+
+# def kill_sim(process):
```
Comment (lukehsiao) [inline run_sim.py:30]: Can get rid off these old commented-out code blocks once we're confident in the new stuff.

File: run_sim.py [hunk 3]
```diff
@@ -74,14 +85,44 @@ def run_sim(args, numHosts=None):
...
+        print "\n[info] Saying yes to first time connection."
+        p.sendline("yes")
+        i = p.expect([ssh_newkey, ".*password:", pexpect.EOF, pexpect.TIMEOUT], timeout=2)
+    if i == 1:
+        print "\n[info] Providing user password."
+        p.sendline(KEY)
+        j = p.expect([".*denied.*", pexpect.EOF, pexpect.TIMEOUT], timeout=2)
+        if j == 0:
+            sys.exit("\n[ERROR] Did you fill in your password?")
+        elif j == 3:
```
Comment (lukehsiao) [inline run_sim.py:107]: Sometimes mav-2 doesn't respond with a permission denied within this timeout window. Can look at making this a little more robust.

**Review by lukehsiao — COMMENTED** [review]
Ultimately, this whole PR is just a work-around for not having passwordless SSH though. So, if we can get that working, let's just discard this PR entirely.

**Comment by kkaffes** [comment]
Since we have the limited output problem with pexpect, I believe we should keep a copy of the old implementation for debugging.

File: run_sim.py (hunk 2 above)
Comment (lukehsiao) [reply run_sim.py:30]: Per @kkaffes comment below, we can leave these in.

**Pull request merged**